src/transformers/models/clipseg/processing_clipseg.py

```diff
@@ -96,20 +113,43 @@ def __call__(self, text=None, images=None, visual_prompt=None, return_tensors=No
               `None`).
             - **pixel_values** -- Pixel values to be fed to a model. Returned when `images` is not `None`.
         """
+
+        output_kwargs = self._merge_kwargs(
+            CLIPSegProcessorKwargs,
+            tokenizer_init_kwargs=self.tokenizer.init_kwargs,
+            **kwargs,
+        )
+
+        if output_kwargs["text_kwargs"].get("visual_prompt") is not None and audio is not None:
+            raise ValueError(
+                "You cannot provide `visual_prompt` as a positional argument and as a keyword argument at the same time."
+                "Please provide it only as a keyword argument (i.e. `visual_prompt=...`)."
+            )
+        if "visual_prompt" not in output_kwargs["text_kwargs"]:
+            warnings.warn(
+                "No `visual_prompt` kwarg was detected. The use of `visual_prompt` as an argument without specifying it explicitely as `visual_prompt=` will be deprecated in future versions."
+            )
+            # For backwards compatibility, we reuse `audio` as `visual_prompt` in case
+            # downstream users passed it as a positional argument
+            if audio is not None:
+                output_kwargs["text_kwargs"]["visual_prompt"] = audio
+
+        visual_prompt = output_kwargs["text_kwargs"].pop("visual_prompt", None)
```

Comment:
same remark as in #32841 , we should not be using kwargs for a purpose other than the one advertised!

Reply:
@molbap , I just moved the models w/ special arg names to this PR: https://github.com/huggingface/transformers/pull/32841 so they'd be in one place.

This PR should now be ready for review.